src/index.js

```diff
@@ -93,31 +84,29 @@ class LndEngine {
   async validateEngine () {
     try {
       const payload = { symbol: this.symbol }
-      const errorMessage = 'Engine failed to validate. Retrying'
       const validationCall = async () => {
-        // An Engine is `locked` when no wallet is present OR if LND Engine requires
-        // a password to unlock the current wallet
-        //
-        // We make this initial call to check if engine is unlocked before continuing
-        // to validate the current engine
-        this.unlocked = await this.isEngineUnlocked()
+        // A macaroon file for lnd will only exist if lnrpc (Lightning RPC) has been started
+        // on LND. Since an engine can become unlocked during any validation call, we
+        // need to ensure that we are updating LndEngine's client in the situation that
+        // a macaroon file becomes available (meaning an engine has been unlocked).
+        this.client = generateLightningClient(this)
 
-        if (!this.unlocked) {
-          throw new Error('LndEngine is locked, unable to validate config')
-        }
+        // Returns a status `ENGINE_STATUS`
+        this.status = await this.getStatus()
 
-        // Once the engine is unlocked, we will attempt to validate our engine's
-        // configuration. If we call `isNodeConfigValid` before the engine is unlocked
-        // the call will fail without a friendly error
-        this.validated = await this.isNodeConfigValid()
+        if (!this.validated) {
```

Comment:
where is `this.validated` being set?

Reply:
forgot to checkin the `getter` for this property